### PR TITLE
chore: improve nse debugging

### PR DESCRIPTION
**Pull request by johnxnguyen**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We still have issues with the NSE not displaying notifications for periods of time. It's hard to know where the failure is happening because 1) notifications are just difficult to debug without a debugger, 2) the NSE is a separate process and our logging tools don't work in it, 3) it not easy to reproduce the issue.

### Solutions

Add two developer flags:

1. `nseDebugEntryPoint`, if on, will make the NSE display the contents of the push it receives at its entry point immediately and then return. This is useful to check whether the problem is if the NSE is receiving the push at all (eg. due to bad network) or if it's to do with our processing of the push.2. 
2. `nseDebugging`, if on, will display notifications with debug messages in all the failure points of the push processing, which would have otherwise resulted in a silent failure. This will be helpful to isolate specific code paths.3. 

### Notes 

These are just some attempt to isolate the cause(s) of the issues we're experiencing. Even if the debug notifications aren't being displayed, but we're sure that the entry point is being hit, then it would suggest something else going wrong such as memory budget violations or crashes.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8013,7 +8015,6 @@
 				5E144FFF213EA0EC00B65E14 /* PreviewImageView.swift in Sources */,
 				5EDD2538213FD5E200C0E254 /* SLComposeServiceViewController+Preview.swift in Sources */,
 				06F9864625CC146B005572C5 /* UIView+ConstraintFactory.swift in Sources */,
-				06717E5D2865183600AF442C /* Stylable.swift in Sources */,
```
Comment (agisilaos) [inline Wire-iOS.xcodeproj/project.pbxproj:8016]: @johnxnguyen Not sure why this keeps happening, but we shouldn't delete those files. Happened in my PR too.

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1218,7 +1218,6 @@
 		DBF7235F1B5566B80036F4C5 /* a_lot_of_people.json in Resources */ = {isa = PBXBuildFile; fileRef = DBF7235E1B5566B80036F4C5 /* a_lot_of_people.json */; };
 		E52D5A7B2858C18600D0494E /* Colors+Generated.swift in Sources */ = {isa = PBXBuildFile; fileRef = E52D5A7A2858C18600D0494E /* Colors+Generated.swift */; };
 		E52F448B2844E916008E11CB /* UITableViewCell+DisclosureIndicator.swift in Sources */ = {isa = PBXBuildFile; fileRef = E52F448A2844E916008E11CB /* UITableViewCell+DisclosureIndicator.swift */; };
-		E5436FAE285A1C83004D95EB /* SemanticColors.swift in Sources */ = {isa = PBXBuildFile; fileRef = E56CA3762858A36500CD2045 /* SemanticColors.swift */; };
```
Comment (agisilaos) [inline Wire-iOS.xcodeproj/project.pbxproj:1221]: @johnxnguyen Not sure why this keeps happening, but we shouldn't delete those files. Happened in my PR too.

Comment (johnxnguyen) [reply Wire-iOS.xcodeproj/project.pbxproj:1221]: I think this file was moved from Wire-iOS to WireCommonComponents. I checked the project and it exists in common components, so I don't think this is a problem.

Comment (agisilaos) [reply Wire-iOS.xcodeproj/project.pbxproj:1221]: I don't think that's correct. I can see the file in Wire-iOS and not in WireCommonComponents. 🤔

Comment (johnxnguyen) [reply Wire-iOS.xcodeproj/project.pbxproj:1221]: You're right, the files are in Wire-iOS. The it's likely the other way around.... that these are "deleted" from WireCommonComponent.

**Review by agisilaos — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**